sonic_installer/main.py

```diff
@@ -388,11 +419,12 @@ def upgrade_docker(container_name, url, cleanup_image, enforce_check, tag):
     DEFAULT_IMAGE_PATH = os.path.join("/tmp/", image_name)
     if url.startswith('http://') or url.startswith('https://'):
         click.echo('Downloading image...')
+        validate_url_or_abort(url)
         try:
             urllib.urlretrieve(url, DEFAULT_IMAGE_PATH, reporthook)
         except Exception, e:
             click.echo("Download error", e)
-            return
+            raise click.Abort()
         image_path = DEFAULT_IMAGE_PATH
     else:
         image_path = os.path.join("./", url)
```

Comment:
Could you  also put the same check after docker image_path is set? Thanks!

```
        if not os.path.isfile(image_path):
            click.echo("Image file does not exist or is not a regular file. Aborting...")
            raise click.Abort()

            # TODO: Validate file is a proper SONiC docker image file
```

Reply:
Done.